server/src/main/java/org/opensearch/common/settings/Setting.java

```diff
@@ -1256,13 +1256,24 @@ private static boolean isFiltered(Property[] properties) {
     public static class RegexValidator implements Writeable, Validator<String> {
         private Pattern pattern;
 
+        private boolean isMatching = true;
+
         /**
          * @param regex A regular expression containing the only valid input for this setting.
          */
         public RegexValidator(String regex) {
             this.pattern = Pattern.compile(regex);
         }
 
+        /**
+         * @param regex A regular expression containing the only valid input for this setting.
+         * @param isMatching this is a boolean switching between "fail, if it matches" or "fail, if it doesn't match".
```

Comment:
This javadoc doesn't tell me which is which.   It should be clear what true means and what false means.  

Something like "If true, the setting must match the given regex. If false, the setting must not match the given regex."